### PR TITLE
Amina/Payments page

### PR DESCRIPTION
**Pull request by amina-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [x] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by reza-fs — DISMISSED** [review]

File: src/pages/payment-methods/index.js [hunk 1]
```diff
@@ -82,14 +84,87 @@ const InnerDiv = styled.div`
...
-    return (
-        <Accordion has_single_state>
-            {payment_data.map((pd, idx) => {
-                if (pd.is_crypto && is_eu_country) {
-                    return []
-                } else
+    const { crypto_config } = React.useContext(DerivStore)
+    const current_client_country = getClientCountry()
+    // eslint-disable-next-line no-console
+    console.log(current_client_country)
```
Comment (prince-deriv) [inline src/pages/payment-methods/index.js:145]: Let's remove the console here

File: src/pages/payment-methods/index.js [hunk 2]
```diff
@@ -82,14 +84,79 @@ const InnerDiv = styled.div`
...
+                if (excludedCountries.includes(current_client_country)) {
+                    return false
+                }
+                if (excludedCountries.includes('eu')) {
+                    return !isEuCountry(current_client_country)
+                }
+                return true
+            }
+            return true
+        })
```
Comment (sean-binary) [inline src/pages/payment-methods/index.js:126]: Can we simplify the if-else block here?

File: src/pages/payment-methods/index.js [hunk 3]
```diff
@@ -82,14 +84,79 @@ const InnerDiv = styled.div`
...
-                } else
+    const { crypto_config } = React.useContext(DerivStore)
+    const current_client_country = getClientCountry()
+    const [show_table, setShowTable] = useState(false)
+
+    const paymentsBasedOnCountry = getPaymentsBasedOnCountry(current_client_country)
+
+    if (!paymentsBasedOnCountry.length) {
+        return (
+            <>
```
Comment (sean-binary) [inline src/pages/payment-methods/index.js:144]: You don't need fragments if it's just one parent component.

File: src/pages/payment-methods/index.js (hunk 2 above)
Comment (amina-deriv) [reply src/pages/payment-methods/index.js]: Moved the logic into two separate functions.
```
const filterCountry = (countriesArray, client_country) => {
    const countries = countriesArray.map((i) => i.toLowerCase())
    if (countries.includes(client_country)) {
        return true
    } else if (countries.includes('eu')) {
        return isEuCountry(client_country)
    }
    return false
}

const getPaymentsBasedOnCountry = () => {
    const client_country = getClientCountry()
    const payments_data_array = payment_data.map((category) => {
        const filteredData = category.data.filter((item) => {
            if (item.countries.included.length)
                return filterCountry(item.countries.included, client_country)
            else if (item.countries.excluded.length)
                return !filterCountry(item.countries.excluded, client_country)
            return true
        })
        return {
            ...category,
            data: filteredData,
        }
    })
    return payments_data_array.filter((item) => item.data.length)
}
```

File: src/pages/payment-methods/index.js (hunk 3 above)
Comment (amina-deriv) [reply src/pages/payment-methods/index.js]: done

File: src/pages/payment-methods/payment-methods-translation.js [hunk 4]
```diff
@@ -0,0 +1,398 @@
+import React from 'react'
+import { localize } from 'components/localization'
```
Comment (prince-deriv) [inline src/pages/payment-methods/payment-methods-translation.js:2]: @sean-binary  this is the file generated by the payment_method script to get handled by translation extractor script

File: src/pages/payment-methods/index.js [hunk 5]
```diff
@@ -32,18 +43,38 @@ const Th = styled.th`
...
+`
+const SectionContentContainer = styled(SectionContainer)`
+    @media ${device.tabletL} {
+        padding: 24px 0 40px;
+    }
+`
+const TopContainer = styled(Container)`
+    @media ${device.tabletL} {
+        width: 100%;
+    }
```
Comment (sean-binary) [inline src/pages/payment-methods/index.js:71]: We can pass all these tabletL styles in as props for e.g. `<SectionContainer tabletL={{ p: "48px 16px 40px" }}></SectionContainer>`.  You can refer to the base component for the responsive styles config for ref.

**Review by sean-binary — APPROVED** [review]
LGTM 👍

File: src/components/hooks/use-browser-resize.js [hunk 6]
```diff
@@ -2,9 +2,14 @@ import { useState, useEffect } from 'react'
 import { isBrowser } from 'common/utility'
 import { size } from 'themes/device'
 
-export const useBrowserResize = () => {
+export const useBrowserResize = (maxSize) => {
     const [is_mobile, setMobile] = useState(false)
     const [width, setWidth] = useState(isBrowser() ? window.innerWidth : 0)
+    const breakpoint = maxSize ? maxSize : size.tablet
```
Comment (prince-deriv) [inline src/components/hooks/use-browser-resize.js:8]: Let's use a shorthand for this

```
const breakpoint = maxSize || size.tablet
```

Comment (amina-deriv) [reply src/components/hooks/use-browser-resize.js]: Done.:+1:

**Review by prince-deriv — DISMISSED** [review]

File: src/pages/payment-methods/index.js (hunk 5 above)
Comment (amina-deriv) [reply src/pages/payment-methods/index.js]: Thanks for the feedback Sean. I have changed the TopContainer style based on this. However sectionContainer has some conflicts in style and waiting to get fixed in this card - https://redmine.deriv.cloud/issues/51208#Section_Container_component_enhancement

File: src/components/containers/container.js [hunk 7]
```diff
@@ -29,7 +29,7 @@ const Container = styled(Box)`
         max-width: 1600px;
     }
     @media ${device.tabletL} {
-        width: 90%;
+        width: ${(props) => props.width || '90%'};
```
Comment (sean-binary) [inline src/components/containers/container.js:32]: I'm afraid that this might affect all other pages globally unintentionally. @amina-deriv

**Pull request closed without merging**